bam-index/src/main/java/org/victorchang/KeyPointerWriter.java

```diff
@@ -31,25 +33,23 @@ public KeyPointerWriter() {
 
     @SuppressWarnings("UnstableApiUsage")
     public List<KeyPointer> write(OutputStream outputStream, Stream<KeyPointer> stream, long blockSize) throws IOException {
-        GzipConcatenatedOutputStream concatenatedStream =
-                new GzipConcatenatedOutputStream(outputStream, 1 << 16, compressionLevel);
+        BlockCompressedOutputStream blockCompressedOutputStream = new BlockCompressedOutputStream(outputStream, (File) null, compressionLevel);
```

Comment:
I haven't look at the code of htsjdk `BlockCompressedOutputStream`. I feel too risky to use it right away without running some tests first.